docs/content/how-to/aws/create-aws-hosted-cluster-spot-instances.md

```diff
@@ -0,0 +1,216 @@
+
+Choose the desired target release image name ([release controller](https://openshift-release.apps.ci.l2s4.p1.openshiftapps.com/)).
+
+```sh
+HOSTED_CLUSTER_NAME=${CLUSTER_PREFIX}-wl
+OCP_RELEASE_IMAGE=<CHANGE_ME_TO_LATEST_RELEASE_IMAGE>
+# Example of image: quay.io/openshift-release-dev/ocp-release:4.19.0-rc.5-x86_64
+
+./hypershift create cluster aws \
+  --name="${HOSTED_CLUSTER_NAME}" \
+  --region="${AWS_REGION}" \
+  --zones="${AWS_REGION}a" \
+  --node-pool-replicas=1 \
+  --base-domain="${CLUSTER_BASE_DOMAIN}" \
+  --pull-secret="${PULL_SECRET_FILE}" \
+  --aws-creds="${AWS_CREDS}" \
+  --ssh-key="${SSH_PUB_KEY_FILE}" \
+  --release-image="${OCP_RELEASE_IMAGE}"
+```
+
+Check the cluster information:
+
+```sh
+oc get --namespace clusters hostedclusters
+oc get --namespace clusters nodepools
+```
+
+When completed, extract the credentials for workload cluster:
+
+```sh
+./hypershift create kubeconfig --name ${HOSTED_CLUSTER_NAME} > kubeconfig-${HOSTED_CLUSTER_NAME}
+
+# kubeconfig for workload cluster
+export KUBECONFIG=$PWD/kubeconfig-${HOSTED_CLUSTER_NAME}
+```
+
+## Configure NodePool with Spot instances
+
+### Option 1: Create a new NodePool with Spot instances
+
+Create a new NodePool configured to use AWS Spot instances:
+
+```sh
+cat << EOF | oc apply -f -
+apiVersion: hypershift.openshift.io/v1beta1
+kind: NodePool
+metadata:
+  name: spot-nodepool
+  namespace: clusters
+spec:
+  clusterName: ${HOSTED_CLUSTER_NAME}
+  replicas: 2
+  management:
+    autoRepair: true
+    upgradeType: Replace
+  platform:
+    aws:
+      instanceType: m5.large
+      subnet:
+        id: subnet-xxxxxxxxx  # Replace with your subnet ID
+      spotMarketOptions:
+        maxPrice: "0.10"  # Maximum price per hour in USD (optional)
+  release:
+    image: ${OCP_RELEASE_IMAGE}
+EOF
+```
+
+### Option 2: Update existing NodePool to use Spot instances
+
+You can also modify an existing NodePool to use Spot instances:
+
+```sh
+oc patch nodepool <nodepool-name> -n clusters --type='merge' -p='
+{
+  "spec": {
+    "platform": {
+      "aws": {
+        "spotMarketOptions": {
+          "maxPrice": "0.10"
+        }
+      }
+    }
+  }
+}'
+```
+
+## Spot instance configuration options
+
+The `spotMarketOptions` field supports the following configuration:
+
+- **maxPrice** (optional): The maximum price per hour that you're willing to pay for a Spot instance, specified in USD. If omitted, the On-Demand price is used as the maximum price.
+
+## Understanding Spot instance behavior
+
+### Cost savings
+
+AWS Spot instances can provide significant cost savings, often 50-90% less than On-Demand pricing, depending on the instance type and availability.
+
+### Availability and interruptions
+
+- Spot instances are subject to interruption when AWS needs the capacity back for On-Demand customers
+- Instances receive a 2-minute warning before termination
+- HyperShift automatically handles instance replacement when Spot instances are terminated
```

Comment:
>- HyperShift automatically handles instance replacement when Spot instances are terminated

can you articulate how?
we will also need to ship a termination handler for detecting upcoming interruptions and do a best effort to handle them gracefully

Reply:
My understanding is that due to auto repair setting part of the node pool management when AWS terminates the spot instance it will cause a failure in the cluster infra which in turn will cause the node pool management to replace it trying to maintain the desired replica count. But for a more graceful approach it would indeed require more integration for detecting interruptions